agile-development/collaboration/Virtual_Collaboration.md

```diff
@@ -0,0 +1,66 @@
+
+Red Team Testing has the same philosophy as any other Test-Driven Development lifecycle: All implementation is divorced from the interface, and the interface can be tested with no knowledge of the implementation.
+
+![ptt-diagram](PTTdiagram.PNG)
+
+## Steps
+
+1. Design Phase: Both programmers will design the interface together. This includes:
+    * Method signatures and names
+    * Writing documentation or docstrings for what the methods are intended to do.
+    * Architecture decisions that would influence testing (Factory patterns, etc.)
+
+2. Implementation Phase: The programmers will separate and parallelize work, while continuing to communicate.
+    * Programmer A will design the implementation of the methods, adhering to the previously decided design.
+    * Programmer B will concurrently write tests for the same method signatures, without knowing details of the implementation.
+
+3. Integration & Testing Phase: Both programmers will commit their code and run the tests.
+    * Utopian Scenario: All tests run and pass correctly.
+    * Realistic Scenario: The tests have either broken or failed due to flaws in testing. This leads to further clarification of the design and a discussion of why the tests failed.
+
+4. The programmers will repeat the three phases until the code is functional and tested.
+
+## When to follow the RTT strategy
+
+RTT works well under specific circumstances. If collaboration needs to happen virtually, and all communication is virtual, RTT reduces the need for constant communication while maintaining the benefits of a joint design session. This considers the human element: Virtual communication is more exhausting than in person communication.
+
+RTT also works well when there is complete consensus, or no consensus at all, on what purpose the code serves. Since creating the design jointly and agreeing to implement and test against it are part of the RTT method, RTT forcibly creates clarity through iteration and communication.
+
+## Benefits
+
+RTT has many of the same benefits as Pair Programming and Test-Driven development but tries to update them for a virtual setting.
+
+* Code implementation and testing can be done in parallel, over long distances or across time zones, which reduces the overall time taken to finish writing the code.
+
+* RTT maintains the pair programming paradigm, while reducing the need for video communication or constant communication between engineers.
+
+* RTT allows detailed focus on design and engineering alignment before implementing any code, leading to cleaner and simpler interfaces.
+
+* RTT forces testing to be prioritized alongside implementation, instead of having testing follow or be influenced by the implementation of the code.
+
+* Documentation is inherently a part of RTT, since both the implementer and the tester need correct, up to date documentation in the implementation phase.
```

Comment:
```suggestion
* Documentation is inherently a part of RTT, since both the implementer and the tester need correct, up to date documentation, in the implementation phase.
```